4/jsschool-2023-main/client/src/scenes/Create/index.jsx

```diff
@@ -0,0 +1,109 @@
+import { memo, useState, useEffect } from "react";
+import axios from "axios";
+import Button from "../../components/Button";
+import styles from "./Create.module.css";
+import Input from "../../components/Input";
+import { useNavigate, useLocation, useParams } from "react-router-dom";
+
+const Create = () => {
+  const location = useLocation();
+  const { robotId } = useParams();
+  const [currentRobot, setCurrentRobot] = useState(0);
+  const [name, setName] = useState();
+  const [error, setError] = useState(false);
+  const navigate = useNavigate();
+
+  const robots = [
+    "/robot-1.svg",
+    "/robot-2.svg",
+    "/robot-3.svg",
+    "/robot-4.svg",
+    "/robot-5.svg",
+    "/robot-6.svg",
+    "/robot-7.svg",
+    "/robot-8.svg",
+    "/robot-9.svg",
+    "/robot-10.svg",
+  ];
+
+  /*
+  const fetchLoadRobotId = async () => {
+    try {
+      console.log("api/robot/"+robotId)
+      const url = "api/robot/id/"+robotId
+      const data = await axios.get(url);
+      console.log(data.data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+*/
+  useEffect(() => {
+    if (location.state) {
+      setCurrentRobot(location.state.robotType);
+    }
+  }, [robotId, location.state]);
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    setError(false);
+    try {
+      if (robotId) {
+        const data = {
+          robotId,
+          name,
+          type: currentRobot,
+        };
+        const response= await axios.put("api/robot/update",{ robotId,name, type: currentRobot });
```

Comment:
ID is something that would ideally (best practices) be sent as a path param:
```suggestion
        const response= await axios.put(`api/robot/update/${robotId}`,{ name, type: currentRobot });
```

I know, I said that sensitive information should **always** be sent in body, but for some reason, ID is usually sent as path param. Probably because it's out of context and doesn't mean anything outside our system.